components/micro-gateway-core/src/main/ballerina/src/gateway/filters/subscription_filter.bal

```diff
@@ -43,137 +44,155 @@ public type SubscriptionFilter object {
+                map<json>? customClaims = jwtPayload?.customClaims;
+                if(customClaims is map<json>) {
+                    if(customClaims.hasKey(APPLICATION)) {
+                        payload["application"] = customClaims[APPLICATION];
+                    }
+                    if(customClaims.hasKey(SUBSCRIBED_APIS)) {
+                        payload["subscribedAPIs"] = customClaims[SUBSCRIBED_APIS];
+                    }
+                    if(customClaims.hasKey(CONSUMER_KEY)) {
+                        payload["consumerKey"] = customClaims[CONSUMER_KEY];
+                    }
+                    if(customClaims.hasKey(KEY_TYPE)) {
+                        payload["keytype"] = customClaims[KEY_TYPE];
+                    }
                 }
-                if (decodedPayload.application.tier != null) {
-                    authenticationContext.applicationTier = decodedPayload.application.tier.toString
-                    ();
+                if(jwtPayload?.sub is string) {
+                    payload["sub"] = jwtPayload?.sub;
                 }
-                authenticationContext.subscriber = decodedPayload.application.owner.toString();
-                authenticationContext.consumerKey = decodedPayload.consumerKey.toString();
-                authenticationContext.keyType = decodedPayload.keytype.toString();
-                runtime:getInvocationContext().attributes[KEY_TYPE_ATTR] = authenticationContext.
-                keyType;
-                filterContext.attributes[AUTHENTICATION_CONTEXT] = authenticationContext;
-                return true;
+                decodedPayload = payload;
             }
-            int index = 0;
-            while (index < l) {
-                var subscription = subscribedAPIList[index];
-                if (subscription.name.toString() == apiConfig.name &&
-                    subscription["version"].toString() == apiConfig.apiVersion) {
-                    printDebug(KEY_SUBSCRIPTION_FILTER, "Found a matching subscription with name:" +
-                            subscription.name.toString() + " version:" + subscription["version"].
-                            toString());
+        } else {
+            //If not found in cache decode jwt token and get the payload
+            var jwtPayload = getEncodedJWTPayload(jwtToken);
+            if (jwtPayload is error) {
+                log:printError(jwtPayload.reason(), err = jwtPayload);
+                setErrorMessageToFilterContext(filterContext, API_AUTH_GENERAL_ERROR);
+                sendErrorResponse(caller, request, filterContext);
+                return false;
+            } else {
+                printTrace(KEY_SUBSCRIPTION_FILTER, "Encoded JWT payload: " + jwtPayload);
+                decodedPayload = getDecodedJWTPayload(jwtPayload);
+            }
+        }
+
+        if(decodedPayload is json) {
+            printTrace(KEY_SUBSCRIPTION_FILTER, "Decoded JWT payload: " + decodedPayload.toString());
```

Comment:
Shall we push logic inside this if block to a separate method? If block seems to be too long